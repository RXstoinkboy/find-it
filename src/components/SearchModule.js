import React from 'react';

const SearchModule = () => {
    return (
        <div>
            this is searchModule
        </div>
    );
};

export default SearchModule;