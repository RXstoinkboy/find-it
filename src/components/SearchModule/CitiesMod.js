import React from 'react';
import styles from './styles/ResultsMod.module.scss'
import PropTypes from 'prop-types'

export const CitiesMod = props => {
    return (
        // display the box only if input is focused
        <div className={props.focus && props.city.length > 0 ? [styles.wrapper, styles.wrapperActive].join(' ') : styles.wrapper} >
            <section className={styles.section} >
            {/* list over cites */}
                <ul className={styles.contentList}>
                    {props.cities.map((city, idx) => (
                        <li 
                            onMouseDown={props.handlePickCity}
                            className={styles.contentItem} 
                            key={idx}>{city.name}</li>
                    ))}
                </ul>
            </section>
        </div>
    );
};

export default CitiesMod;

CitiesMod.propTypes = {
    city: PropTypes.string.isRequired,
    focus: PropTypes.bool.isRequired,
    handlePickCity: PropTypes.func
}