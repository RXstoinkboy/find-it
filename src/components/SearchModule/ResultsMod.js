import React from 'react';
import styles from './styles/ResultsMod.module.scss'
import PropTypes from 'prop-types'

export const ResultsMod = props => {
    return (
        // display the box only if input is focused
        <div className={props.focus && props.keywordToSearch.length > 0 ? [styles.wrapper, styles.wrapperActive].join(' ') : styles.wrapper} >
        {/* display list of categories which were found from autocomplete query */}
            {Object.keys(props.autocompleteResults).map(section => (
                <section className={styles.section} key={section}>
                {/* display the title of each category but only if it is not empty */}
                    <h4 className={styles.sectionTitle}>{
                        props.autocompleteResults[section].length > 0 ?
                            section
                                : null
                    }</h4>
                    {/* display content of each category */}
                    <ul className={styles.contentList}>
                        {props.autocompleteResults[section].map((item, idx) => (
                            <li 
                                onMouseDown={props.handlePickWord}
                                value={
                                    section === 'categories' ? item.title
                                        : section === 'businesses' ? item.name
                                            : item.text
                                }
                                className={styles.contentItem} 
                                key={idx}
                            >
                                {
                                    section === 'categories' ? item.title
                                        : section === 'businesses' ? item.name
                                            : item.text
                                }
                            </li>
                        ))}
                    </ul>
                </section>
            ))}

        </div>
    );
};

export default ResultsMod;

ResultsMod.propTypes = {
    focus: PropTypes.bool.isRequired,
    keywordToSearch: PropTypes.string.isRequired,
    autocompleteResults: PropTypes.object,
    handlePickWord: PropTypes.func
}