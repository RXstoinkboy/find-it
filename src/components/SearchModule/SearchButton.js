import React from 'react';
import styles from './styles/SearchButton.module.scss'
import {ReactComponent as SearchIcon} from '../../assets/img/search.svg'

const SearchButton = props => {
    return (
        <button 
            // type='submit'
            className={styles.searchButton}
            onClick={props.handleGetBusinessesList}
        >
            <SearchIcon className={styles.searchIcon}/>
        </button>
    );
};

export default SearchButton;