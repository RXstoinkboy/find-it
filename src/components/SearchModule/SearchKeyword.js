import React from 'react';
import styles from './styles/SearchInput.module.scss'
import PropTypes from 'prop-types'

export const SearchKeyword = props => {
        return (
            <input 
            type="search" 
            placeholder='Search by name, category etc...' 
            onChange={props.handleSearch}
            value={props.keywordToSearch}
            className={styles.searchInput} />
        );
};

export default SearchKeyword;

SearchKeyword.propTypes = {
    handleSearch: PropTypes.func.isRequired,
    keywordToSearch: PropTypes.string.isRequired
}