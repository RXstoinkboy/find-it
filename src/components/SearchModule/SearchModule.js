import React from 'react';
import styles from './styles/SearchModule.module.scss'


export const SearchModule = props => {
    return (
        <form className={styles.wrapper}>
            {props.children}
        </form>
    );
};

export default SearchModule;