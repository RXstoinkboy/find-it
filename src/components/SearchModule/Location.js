import React, {Component} from 'react';
import styles from './styles/Location.module.scss'
import utils from './styles/SearchKeyword.module.scss'
import PropTypes from 'prop-types'

import CitiesMod from './CitiesMod'

export class Location extends Component {
    constructor(props){
        super(props);
        this.state = {
            focus: false
        }
    }

    toggleFocus = e => {
        this.setState({
            focus: !this.state.focus
        })
    }

    render(){
        return (
            <>
                {this.props.filters && <label for='location' className={utils.labelFilters}>place</label>}
                <div className={utils.wrapper}>
                    <input 
                        id='location'
                        type="search" 
                        placeholder='Where are you?' 
                        onChange={this.props.handleSetCity}
                        onFocus={this.toggleFocus}
                        onBlur={this.toggleFocus}
                        className={!this.props.filters ? styles.location : utils.searchInputFilters} 
                        value={this.props.city}/>
                    <CitiesMod {...this.props} focus={this.state.focus} />
                </div>
            </>
        );
    }
};

export default Location;

Location.propTypes = {
    handleSetCity: PropTypes.func.isRequired,
    city: PropTypes.string.isRequired,
    cities: PropTypes.array.isRequired
}