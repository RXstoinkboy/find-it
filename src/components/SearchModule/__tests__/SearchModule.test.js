import React from 'react'
import {SearchModule} from '../SearchModule'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

it(`SearchModule renders correctly`, ()=> {
    const search = shallow(<SearchModule />)

    expect(toJson(search)).toMatchSnapshot()
})