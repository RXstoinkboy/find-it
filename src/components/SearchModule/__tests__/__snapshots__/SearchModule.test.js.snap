// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchModule renders correctly 1`] = `
<form
  className="wrapper"
/>
`;
